blockchain/storage/batch_helper.go

```diff
@@ -0,0 +1,92 @@
+package storage
+
+import (
+	"math/big"
+
+	"github.com/0xPolygon/polygon-edge/helper/common"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/fastrlp"
+)
+
+type Batch interface {
+	Delete(key []byte)
+	Write() error
+	Put(k []byte, data []byte)
+}
+
+type BatchHelper struct {
+	batch Batch
+}
+
+func NewBatchHelper(storage Storage) *BatchHelper {
+	batch := storage.NewBatch()
+
+	return &BatchHelper{batch: batch}
+}
+
+func (b *BatchHelper) WriteHeader(h *types.Header) {
```

Comment:
Maybe we should rename all these WriteXYZ methods to PutXYZ because actual writing to db occurs on `WriteBatch`